src/MergePlugin.php

```diff
@@ -422,6 +429,41 @@ protected function mergeLinks(array $origin, array $merge, array &$dups)
     }
 
     /**
+     * Merge the extras sections of two config files.
+     *
+     * @param RootPackage $root The root package.
+     * @param CompletePackage $package The imported package to merge.
+     */
+    public function mergeExtras(
+        RootPackage $root,
+        CompletePackage $package
+    ) {
+        $this->debug('Merging extra section');
+        $packageExtra = $package->getExtra();
+        if (empty($packageExtra)) {
+            return;
+        }
+
+        if (isset($packageExtra['merge-plugin'])) {
+            $this->debug("Skipping merge-plugin key");
+            unset($packageExtra['merge-plugin']);
+        }
+        $rootExtra = $root->getExtra();
+        foreach ($packageExtra as $key => $value) {
+            $this->debug("Merging extra key <comment>{$key}</comment>");
+            if (isset($rootExtra[$key])) {
```

Comment:
I did think about that but I couldn't think of any other merge behaviours that made sense and I considered safe. We can't use composer's built-in conflict resolution for merging. I guess we might want to optionally enable extras-merging with a flag but maybe not the 'replace' flag. What was your thinking?